jt_wrs/jtracker/job.py

```diff
@@ -26,8 +26,16 @@ def _add_missing_required_param(self):
                 if 'default' in workflow_input_def[key]:  # only use default when it's defined
                     if key not in self.job_json:   # only add to job json when the param is not defined
                         self.job_json[key] = workflow_input_def[key]['default']
-                        if pattern.match(self.job_json[key]):  # we check whether to insert system workflow data path
-                            self.job_json[key] = pattern.sub(r'[${_wf_data}/\1]\2', self.job_json[key])
+                        if isinstance(self.job_json[key], str):
+                            if pattern.match(self.job_json[key]):  # we check whether to insert system workflow data path
+                                self.job_json[key] = pattern.sub(r'[${_wf_data}/\1]\2', self.job_json[key])
+                        elif isinstance(self.job_json[key], list): # default is a list
+                            for i in range(len(self.job_json[key])):
+                                if not isinstance(self.job_json[key][i], str): continue
+                                if pattern.match(self.job_json[key][i]):
+                                    self.job_json[key][i] = pattern.sub(r'[${_wf_data}/\1]\2', self.job_json[key][i])
+                        else:
+                            print('Error: Unsupported default value in the workflow definition!')
```

Comment:
Don't need the `else` here, if the field's value type is not `str` is not an error, we just skip the replacement of `${_wf_data}` since it's not applicable.

Reply:
this `else` means when field's value is not `str` or `list`.

Reply:
correct, it's perfect to do nothing about it if it's not `str`.